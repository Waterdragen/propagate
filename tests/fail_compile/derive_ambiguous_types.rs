use propagate::Propagate;

#[derive(Propagate)]
enum MyEnum {
    #[good]
    A(i32, i32),
    #[good]
    B((i32, i32)),
}

fn main() {}
