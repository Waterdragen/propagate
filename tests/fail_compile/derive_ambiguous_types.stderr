error: Types `(i32, i32)` and `i32, i32` are ambiguous. Cannot infer types for both tuple and n-args variants.
 --> tests/fail_compile/derive_ambiguous_types.rs:3:10
  |
3 | #[derive(Propagate)]
  |          ^^^^^^^^^
  |
  = note: this error originates in the derive macro `Propagate` (in Nightly builds, run with -Z macro-backtrace for more info)
